daemon/build.go

```diff
@@ -14,3 +15,12 @@ import (
 func (daemon *Daemon) ImageExportedByBuildkit(ctx context.Context, id string, desc ocispec.Descriptor) {
 	daemon.imageService.LogImageEvent(id, id, events.ActionCreate)
 }
+
+// ImageNamedByBuildkit is a callback that is called when an image is tagged by buildkit.
+// Note: It is only called if the buildkit didn't call the image service itself to perform the tagging.
+// Currently this only happens when the containerd image store is used.
+func (daemon *Daemon) ImageNamedByBuildkit(ctx context.Context, ref reference.NamedTagged, desc ocispec.Descriptor) {
+	id := desc.Digest.String()
+	name := reference.FamiliarString(ref)
+	daemon.imageService.LogImageEvent(id, name, events.ActionTag)
+}
```

Comment:
See my previous comment; if we had event-type as argument; all of these could effectively be a single method that issues image events (I would change the `ref` argument to be a string and keep it generic (`nameOrID`, or could still be just `ref` as both are references as well).